waspc/packages/wasp-config/__tests__/testFixtures.test-d.ts

```diff
@@ -0,0 +1,95 @@
+
+describe("MinimalConfig<T>", () => {
+  test("should not affect primitive types", async () => {
+    expectTypeOf<MinimalConfig<string>>().toEqualTypeOf<string>();
+    expectTypeOf<MinimalConfig<number>>().toEqualTypeOf<number>();
+    expectTypeOf<MinimalConfig<boolean>>().toEqualTypeOf<boolean>();
+    expectTypeOf<MinimalConfig<null>>().toEqualTypeOf<null>();
+    expectTypeOf<MinimalConfig<undefined>>().toEqualTypeOf<undefined>();
+    expectTypeOf<MinimalConfig<bigint>>().toEqualTypeOf<bigint>();
+    expectTypeOf<MinimalConfig<symbol>>().toEqualTypeOf<symbol>();
+  });
+
+  test("should not affect branded types", async () => {
+    expectTypeOf<MinimalConfig<BrandedType>>().toEqualTypeOf<BrandedType>();
+  });
+
+  test("should convert no props object to EmptyObject", async () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-object-type
+    expectTypeOf<MinimalConfig<{}>>().toEqualTypeOf<EmptyObject>();
+  });
+
+  test("should remove optional props", async () => {
+    expectTypeOf<MinimalConfig<OptionalObject>>().toEqualTypeOf<EmptyObject>();
+  });
+
+  test("should not affect required props", async () => {
+    expectTypeOf<
+      MinimalConfig<RequiredObject>
+    >().toEqualTypeOf<RequiredObject>();
+  });
+
+  test("should handle objects recursively", async () => {
+    expectTypeOf<MinimalConfig<{ nested: unknown }>>().toEqualTypeOf<{
+      nested: MinimalConfig<unknown>;
+    }>();
+  });
+
+  test("should handle objects recursively", async () => {
+    expectTypeOf<MinimalConfig<unknown[]>>().toEqualTypeOf<
+      MinimalConfig<unknown>[]
+    >();
+  });
+});
+
+describe("FullConfig<T>", () => {
+  test("should not affect primitive types", async () => {
+    expectTypeOf<FullConfig<string>>().toEqualTypeOf<string>();
+    expectTypeOf<FullConfig<number>>().toEqualTypeOf<number>();
+    expectTypeOf<FullConfig<boolean>>().toEqualTypeOf<boolean>();
+    expectTypeOf<FullConfig<null>>().toEqualTypeOf<null>();
+    expectTypeOf<FullConfig<undefined>>().toEqualTypeOf<undefined>();
+    expectTypeOf<FullConfig<bigint>>().toEqualTypeOf<bigint>();
+    expectTypeOf<FullConfig<symbol>>().toEqualTypeOf<symbol>();
+  });
+
+  test("should not affect branded types", async () => {
+    expectTypeOf<FullConfig<BrandedType>>().toEqualTypeOf<BrandedType>();
+  });
+
+  test("should not affect required props", async () => {
+    expectTypeOf<FullConfig<RequiredObject>>().toEqualTypeOf<RequiredObject>();
+  });
+
+  test("should make optional props required", async () => {
+    expectTypeOf<FullConfig<OptionalObject>>().toEqualTypeOf<RequiredObject>();
+  });
+
+  test("should not affect required props", async () => {
+    expectTypeOf<FullConfig<RequiredObject>>().toEqualTypeOf<RequiredObject>();
+  });
+
+  test("should handle objects recursively", async () => {
+    expectTypeOf<FullConfig<{ nested: unknown }>>().toEqualTypeOf<{
+      nested: FullConfig<unknown>;
+    }>();
+  });
```

Comment:
We're missing a test for optional nested properties. Same with `MinimalConfig`.

You can do it in the same test - one prop is optional, the other isn't.